plugins/org.jboss.reddeer.core/src/org/jboss/reddeer/core/handler/WidgetHandler.java

```diff
@@ -74,6 +74,22 @@ public boolean isEnabled(Widget widget) {
 		}
 		return ret;
 	}
+	
+	/**
+	 * Finds out whether specified {@link Widget} is disposed or not.
+	 * 
+	 * @param widget widget to handle
+	 * @return true if widget is disposed, false otherwise
+	 */
+	public boolean isDisposed(final Widget widget) {
+		return Display.syncExec(new ResultRunnable<Boolean>() {
+
+			@Override
+			public Boolean run() {
```

Comment:
That's exactly the point. Sometimes Eclipse dispose some widgets even there is no reason for it. We had couple of issues with it already. So when user is asking if widget is disposed widget is still valid but inside syncExec() variable is set to null. I'm not sure if this can happen but from user point of view if widget is null is for sure disposed